cg/services/orders/store_order_services/store_case_order.py

```diff
@@ -34,263 +38,182 @@ def __init__(
-        link_obj = self.status_db.relate_sample(
-            case=case_obj,
-            sample=family_samples[sample["name"]],
-            status=sample["status"],
-            mother=mother_obj,
-            father=father_obj,
+    def _update_relationship(
+        father: DbSample | None, link: CaseSample, mother: DbSample | None, sample: SampleInCase
+    ) -> None:
+        link.status = getattr(sample, "status", link.status)
+        link.mother = mother or link.mother
+        link.father = father or link.father
+
+    def _create_link(
+        self,
+        case: DbCase,
+        family_samples: dict,
+        father: DbSample,
+        mother: DbSample,
+        sample: SampleInCase,
+    ) -> CaseSample:
+        link = self.status_db.relate_sample(
+            case=case,
+            sample=family_samples[sample.name],
+            status=getattr(sample, "status", None),
+            mother=mother,
+            father=father,
         )
-        self.status_db.session.add(link_obj)
-        return link_obj
+        self.status_db.session.add(link)
+        return link
 
-    def _create_sample(self, case, customer_obj, order, ordered, sample, ticket):
-        sample_obj = self.status_db.add_sample(
-            name=sample["name"],
-            comment=sample["comment"],
-            control=sample["control"],
-            internal_id=sample["internal_id"],
-            order=order,
+    def _create_sample(
+        self,
+        case: Case,
+        customer: Customer,
+        order_name: str,
+        ordered: datetime,
+        sample: SampleInCase,
+        ticket: str,
+    ):
+        db_sample: DbSample = self.status_db.add_sample(
+            internal_id=sample._generated_lims_id,
+            order=order_name,
             ordered=ordered,
             original_ticket=ticket,
-            tumour=sample["tumour"],
-            age_at_sampling=sample["age_at_sampling"],
-            capture_kit=sample["capture_kit"],
-            phenotype_groups=sample["phenotype_groups"],
-            phenotype_terms=sample["phenotype_terms"],
-            priority=case["priority"],
-            reference_genome=sample["reference_genome"],
-            sex=sample["sex"],
-            subject_id=sample["subject_id"],
+            priority=case.priority,
+            **sample.model_dump(exclude={"application", "container", "container_name"}),
         )
-        sample_obj.customer = customer_obj
+        db_sample.customer = customer
         with self.status_db.session.no_autoflush:
-            application_tag = sample["application"]
-            sample_obj.application_version: ApplicationVersion = (
-                self.status_db.get_current_application_version_by_tag(tag=application_tag)
+            application_tag = sample.application
+            db_sample.application_version = self.status_db.get_current_application_version_by_tag(
+                tag=application_tag
             )
```

Comment:
the customer and application version can be added as parameter of the `self.status_db.add_sample` function

Reply:
and the autoflush can be called in the outer function